internal/expr/query.go

```diff
@@ -89,7 +89,87 @@ func (pe *PreparedExpr) Query(args ...any) (ce *QueryExpr, err error) {
 	return &QueryExpr{outputs: pe.outputs, sql: pe.sql, args: qargs}, nil
 }
 
-// ScanArgs returns list of pointers to the struct fields that are listed in qe.outputs.
+// sliceInfo contains information about a column slice passed to All.
```

Comment:
We certainly could do it like that I believe. The main reasons for having the bulk of the code in the `query.go` and using the `AllInfo` are
- We avoid the unnecessary type checks at the start of `ScanArgs` by not going through `Decode`
- The AllInfo struct is useful for saving info about the slices types that we then don't need to then generate again
- We keep the `All` in `sqlair.go` short and keep the use of `reflect` confined to `expr`

Ultimately the current implementation pretty much does the same things in the same order as above but tries to avoid lots of extra reflect.ValueOf/type checking.

Reply:
It's not worth buying logic that is significantly more complex before we can even measure the actual impact of the two implementations, and once we do get to the point of optimizing, we should focus on optimizing Iterator itself instead of All.

Reply:
Ok, I'll have a go at implementing as above.

Reply:
Changes made!